src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -1735,29 +1748,20 @@ bool OpFusionHelper::isControlFlowValidForFusion(
 // assumed the canonicalization has hoisted all constant to the beginning of the
 // function by fold function.
 bool OpFusionHelper::areInputsValidForFusion(
-    Operation *useOp, Operation *defOp) {
+    Operation *useOp, Operation *defOp, DimAnalysis *dimAnalysis) {
   // Elementwise unary operation is always fusible
   if (useOp->getOperands().size() == 1)
     return true;
 
-  // To fuse Elementwise op with more one operands with the producer,
-  // the shape of the output the user Op has to have the same size
-  // output as that of the producer Op. Here dimension expansion with size
-  // 1 is allowed. Refer to hasNoBroadcast() definition.
-  // ToFix: This PR simply check static shape and does not use symbolic
-  // shape analysis and BroadcastShapeHelper
-  // Some discussion can be found at
-  // https://github.com/onnx/onnx-mlir/issues/2199
-
-  if (!hasStaticShape(defOp->getResults()[0].getType()))
-    return false;
-
-  ArrayRef<int64_t> defShape = getShape(defOp->getResults()[0].getType());
-  ArrayRef<int64_t> useShape = getShape(useOp->getResults()[0].getType());
+  Type defOutputType = defOp->getResultTypes()[0];
+  Type useOutputType = useOp->getResultTypes()[0];
+  ArrayRef<int64_t> defShape = getShape(defOutputType);
+  ArrayRef<int64_t> useShape = getShape(useOutputType);
   if (defShape != useShape) {
     return false;
   }
 
+  // Check the inputs in the defOp
```

Comment:
Typo? "Check the inputs in the **useOp**"

Reply:
So one of the constraints is that all the "other" inputs of the candidate useOp are constants? If so, it would be helpful to have a description at the top of the class that lists the current limitations. What if it uses an input of an op that is already in the set of fused op, wouldn't that be ok too?

Reply:
Currently, the other inputs have to be constant or block argument. In theory, it is okay that another input comes from the set of fused op. Just need a DAG to record the fusible candidates, while a list is used now. Will consider this extension.